spacy/gold/corpus.py

```diff
@@ -21,6 +22,36 @@ def create_docbin_reader(
 ) -> Callable[["Language"], Iterable[Example]]:
     return Corpus(path, gold_preproc=gold_preproc, max_length=max_length, limit=limit)
 
+@util.registry.readers("spacy.JsonlReader.v1")
+def create_jsonl_reader(
+    path: Path, min_length: int=0, max_length: int = 0, limit: int = 0
```

Comment:
I'm personally not a fan of using `0` for `min_length`'s default here, why not set it to some high int value so we don't need to treat it specially in the code? Or use `None` instead?

Reply:
If we use `None` we need to make it `Optional`, and then the type analysis is harder. 0 is an invalid maximum, so it makes sense to use it as the "no maximum" value. In code that needs to be C-typed, there's no alternative, it has to be 0 and cannot be `None`.

Using a "very large" value is much uglier imo, there should be a way to specify "no limit".